src/main/java/org/wildfly/security/auth/server/SecurityDomain.java

```diff
@@ -81,6 +83,7 @@
         this.principalDecoder = builder.principalDecoder;
         this.securityIdentityTransformer = builder.securityIdentityTransformer;
         this.trustedSecurityDomain = builder.trustedSecurityDomain;
+        this.scheduledExecutorService = builder.scheduledExecutorService;
```

Comment:
Since we don't support security domain shutdown, maybe it's best to have one single shared one.  It can be stored in a static field in a static nested class, so that it is only lazily instantiated, and only one time. ("lazy holder" pattern).